dds_web/static/swagger.yaml

```
@@ -0,0 +1,656 @@
+  version: "2.6.0"
+  title: Data Delivery System
+# Apply the encrypted token to all operations
+security:
+  - seconFactorToken: []
+servers:
+  # Added by API Auto Mocking Plugin
+  - url: https://delivery.scilifelab.se/api/v1
+    description: Endpoint for the production enviroment
+  - url: https://dds-dev.dckube3.scilifelab.se/api/v1
+    description: Endpoint for the development enviroment
+  # Uncomment when local testing
+#  - url: http://localhost:5000/api/v1
+#    description: Local enviroment
+tags:
+  - name: authentication
+    description: Authorization and authentication operations
+  - name: files
+    description: Manage files operations
+  - name: user
+    description: Operations about user
+  - name: project
+    description: Operations about projects
+  - name: s3
+    description: s3 bucket
+  - name: superadmin
+    description: Superadmin operations
+paths:
+  ### FILES OPERATIONS #########
+  /file/new:
+    post:
+      tags:
+        - files
+      summary: Add a new file to the database
+      operationId: newFile
+      parameters:
+        - $ref: "#/components/parameters/defaultHeader"
+        - $ref: "#/components/parameters/projectID"
+      responses:
+        "401":
+          $ref: "#/components/responses/UnauthorizedToken"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "500":
+          $ref: "#/components/responses/InternalServerlError"
+        "200":
+          description: succesful operation
+          content:
+            application/json:
+              schema:
+                type: string
+                example: "{message: File data_seq.txt added to db.}"
+      requestBody:
+        $ref: "#/components/requestBodies/Files"
+    put:
+      tags:
+        - files
+      summary: Update an existing file
+      operationId: updateFile
+      parameters:
+        - $ref: "#/components/parameters/defaultHeader"
+        - $ref: "#/components/parameters/projectID"
+      responses:
+        "401":
+          $ref: "#/components/responses/UnauthorizedToken"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "500":
+          $ref: "#/components/responses/InternalServerlError"
+        "200":
+          description: succesful operation
+          content:
+            application/json:
+              schema:
+                type: string
+                example: "{message: File data_seq.txt updated in the db.}"
+      requestBody:
+        $ref: "#/components/requestBodies/Files"
+  /file/match:
+    get:
+      tags:
+        - files
+      summary: Get name in bucket for all files specified CHECK METHOD
+      description: This method requires the data
+        to be passed in the request body instead of the query,
+        because it is not according to the openAPI standar swagger does not
+        let document it properly, therefore it is plan to be
+        removed in the future.
```

a question, just to be sure: we have made the method so that the data is passed in the request body, and openAPI standard requirers the data to be passed as parameters/query. If so, I suggest a minor change in the text:
```suggestion
      description: This method requires the data
        to be passed in the request body instead of the query.
        Since this does not comply with the openAPI standards, swagger cannot document it properly, therefore we need to change/remove it in the future.
```